Circus not working without internet connection

Hello,
When I start `circusd` without an active Internet connection, I have the following output :

```
2014-02-20 15:43:39 [3665] [INFO] Registering signals...
2014-02-20 15:43:39 [3665] [INFO] Starting master on pid 3665
2014-02-20 15:43:39 [3665] [ERROR] exception [Errno 19] No such device caught
  File "/home/yannick/.virtualenvs/onitu/lib/python2.7/site-packages/tornado/gen.py", line 221, in wrapper
    runner.run()
  File "/home/yannick/.virtualenvs/onitu/lib/python2.7/site-packages/tornado/gen.py", line 507, in run
    yielded = self.gen.send(next)
  File "/home/yannick/.virtualenvs/onitu/lib/python2.7/site-packages/circus/arbiter.py", line 481, in start
    self.ctrl.start()
  File "/home/yannick/.virtualenvs/onitu/lib/python2.7/site-packages/circus/controller.py", line 86, in start
    self.initialize()
  File "/home/yannick/.virtualenvs/onitu/lib/python2.7/site-packages/circus/controller.py", line 66, in initialize
    self.udp_socket = create_udp_socket(multicast_addr, multicast_port)
  File "/home/yannick/.virtualenvs/onitu/lib/python2.7/site-packages/circus/util.py", line 734, in create_udp_socket
    socket.inet_aton(mcast_addr) + socket.inet_aton(any_addr))
  File "/usr/lib64/python2.7/socket.py", line 224, in meth
    return getattr(self._sock,name)(*args)
```

Is it expected ? Is there any way I could avoid that ?

--- a/circus/controller.py
+++ b/circus/controller.py
@@ -53,6 +53,20 @@ class Controller(object):
         self.stream = zmqstream.ZMQStream(self.ctrl_socket, self.loop)
         self.stream.on_recv(self.handle_message)
 
+    def _init_multicast_endpoint(self):
+        multicast_addr, multicast_port = urlparse(self.multicast_endpoint)\
+            .netloc.split(':')
+        try:
+            self.udp_socket = create_udp_socket(multicast_addr,
+                                                multicast_port)
+            self.loop.add_handler(self.udp_socket.fileno(),
+                                  self.handle_autodiscover_message,
+                                  ioloop.IOLoop.READ)
+        except (IOError, OSError, ValueError):
+            message = ("Multicast discovery is disabled, there was an"
+                       "error during udp socket creation.")
+            logger.warning(message, exc_info=True)
+
     @property
     def endpoint_owner_mode(self):
         return self.endpoint_owner is not None and \
@@ -76,12 +90,7 @@ class Controller(object):
 
         # Initialize UDP Socket
         if self.multicast_endpoint:
-            multicast_addr, multicast_port = urlparse(self.multicast_endpoint)\
-                .netloc.split(':')
-            self.udp_socket = create_udp_socket(multicast_addr, multicast_port)
-            self.loop.add_handler(self.udp_socket.fileno(),
-                                  self.handle_autodiscover_message,
-                                  ioloop.IOLoop.READ)
+            self._init_multicast_endpoint()
 
     def manage_watchers(self):
         if self._managing_watchers_future is not None:

--- a/circus/tests/test_controller.py
+++ b/circus/tests/test_controller.py
@@ -1,5 +1,9 @@
 from circus.tests.support import TestCase, EasyTestSuite
 from circus.controller import Controller
+from circus.util import DEFAULT_ENDPOINT_MULTICAST
+from circus import logger
+
+import circus.controller
 
 from zmq.eventloop import ioloop
 
@@ -35,5 +39,42 @@ class TestController(TestCase):
         controller.start()
         loop.start()
         self.assertTrue(controller.called)
+
+    def _multicast_side_effect_helper(self, side_effect):
+        arbiter = mock.MagicMock()
+        loop = mock.MagicMock()
+        context = mock.sentinel.context
+
+        controller = circus.controller.Controller(
+            'endpoint', DEFAULT_ENDPOINT_MULTICAST, context, loop, arbiter
+        )
+
+        with mock.patch('circus.util.create_udp_socket') as m:
+            m.side_effect = side_effect
+            circus.controller.create_udp_socket = m
+
+            with mock.patch.object(logger, 'warning') as mock_logger_warn:
+                controller._init_multicast_endpoint()
+                self.assertTrue(mock_logger_warn.called)
+
+    def test_multicast_ioerror(self):
+        self._multicast_side_effect_helper(IOError)
+
+    def test_multicast_oserror(self):
+        self._multicast_side_effect_helper(OSError)
+
+    def test_multicast_valueerror(self):
+        arbiter = mock.MagicMock()
+        loop = mock.MagicMock()
+        context = mock.sentinel.context
+
+        wrong_multicast_endpoint = 'udp://127.0.0.1:12027'
+        controller = Controller('endpoint', wrong_multicast_endpoint,
+                                context, loop, arbiter)
+
+        with mock.patch.object(logger, 'warning') as mock_logger_warn:
+            controller._init_multicast_endpoint()
+            self.assertTrue(mock_logger_warn.called)
+
 
 test_suite = EasyTestSuite(__name__)

